coalib/misc/Shell.py

```diff
@@ -4,6 +4,56 @@
 from subprocess import PIPE, Popen
 from shutil import which
 
+from coala_utils.decorators import enforce_signature
+
+
+class ShellCommandResult(tuple):
+    """
+    The result of a :func:`coalib.misc.run_shell_command` call.
+
+    It is based on a ``(stdout, stderr)`` string tuple like it is returned
+    form ``subprocess.Popen.communicate`` and was originally returned from
+    :func:`coalib.misc.run_shell_command`. So it is backwards-compatible.
+
+    It additionally stores the :attr:`.process` as ``Popen`` instance
+    and provides a :attr:`.code` to get the ``.process.returncode``:
+
+    >>> process = Popen(['python', '-c', 'print("output")'],
+    ...                 stdout=PIPE, stderr=PIPE, universal_newlines=True)
+    >>> stdout, stderr = process.communicate()
+
+    >>> result = ShellCommandResult(process, stdout, stderr)
+    >>> result[0]
+    'output\\n'
+    >>> result[1]
+    ''
+    >>> result.process is process
+    True
+    >>> result.code
+    0
+    """
+
+    @enforce_signature
+    def __new__(cls, process: Popen, stdout: str, stderr: str):
+        """
+        Creates the basic tuple from `stdout` and `stderr`.
+        """
+        return tuple.__new__(cls, (stdout, stderr))
+
+    @enforce_signature
+    def __init__(self, process: Popen, stdout: str, stderr: str):
+        """
+        Stores the `process` instance.
+        """
+        self.process = process
```

Comment:
You might want to get additional information from the process object later... like... hmm... well... do we really need it? :D